avm/ptn/deployment-script/import-image-to-acr/main.bicep

```
@@ -0,0 +1,246 @@
+metadata name = 'import-image-to-acr'
+metadata description = 'This modules deployes an image to an Azure Container Registry.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Required. The name of the deployment script resource.')
+param name string
+
+@description('Optional. Location for all Resources.')
+param location string = resourceGroup().location
+
+@description('Optional. Enable/Disable usage telemetry for module.')
+param enableTelemetry bool = true
+
+@description('Required. The name of the Azure Container Registry.')
+param acrName string
+
+@description('Optional. How the deployment script should be forced to execute. Default is to force the script to deploy the image to run every time.')
+param runOnce bool = false
+
+@description('Optional. The RoleDefinitionId required for the DeploymentScript resource to import images. If not set, an existing managed identity should have the appropriate role assigned. Defaults to Contributor.')
+param rbacRole string = 'b24988ac-6180-42a0-ab88-20f7382dd24c' //Contributor is needed to import ACR
+
+@description('Optional. The managed identity definition for this resource.')
+param managedIdentities managedIdentitiesType?
+
+@description('Conditional. Name of the Managed Identity resource to create. Required if `managedIdentities` is `null`. Defaults to `id-ContainerRegistryImport`.')
+param managedIdentityName string?
+
+@description('Required. A fully qualified image name to import.')
+@metadata({
+  example: 'mcr.microsoft.com/k8se/quickstart-jobs:latest'
+})
+param image string
+
+@description('Optional. The image will be overwritten if it already exists in the ACR with the same tag. Default is false.')
+param overwriteExistingImage bool = false
+
+@description('Optional. A delay in seconds before the script import operation starts. Primarily to allow Azure AAD Role Assignments to propagate. Default is 30s.')
+param initialScriptDelay int = 30
+
+@description('Optional. The maximum number of retries for the script import operation. Default is 3.')
+param retryMax int = 3
+
+@allowed([
+  'OnSuccess'
+  'OnExpiration'
+  'Always'
+])
+@description('Optional. When the script resource is cleaned up. Default is OnExpiration and the cleanup time is after 1h.')
+param cleanupPreference string = 'OnExpiration'
+
+@description('Optional. The resource id of the storage account to use for the deployment script. An existing storage account is needed, if PrivateLink is going to be used for the deployment script.')
+param storageAccountResourceId string = ''
+
+@description('Optional. The subnet ids to use for the deployment script. An existing subnet is needed, if PrivateLink is going to be used for the deployment script.')
+param subnetIds string[]?
```

```suggestion
param subnetResourceIds string[]?
```

Would need to be updated in other locations too

---

The original comment [here](https://github.com/Azure/bicep-registry-modules/pull/2255#discussion_r1661516765) may have been overlooked